client/src/components/books/books/BooksModal.1.js

```diff
@@ -0,0 +1,349 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Document, Page } from 'react-pdf';
+import cloudinary from 'cloudinary';
+import { saveBooks, addBook } from '../../../actions/books';
+import Modal from '../../modal/Modal';
+import Button from '../../form/Button';
+import InputField from '../../form/InputField';
+import Textarea from '../../form/TextareaField';
+import FlashMessagesList from '../../flash/FlashMessagesList';
+import { addFlashMessage } from '../../../actions/flashMessages';
+import UploadBookCover from './UploadBookCover';
+
+const $ = window.$;
+
+/* eslint-disable require-jsdoc, class-methods-use-this */
+class BooksModal extends Component {
+    state = {
+      title: this.props.book ? this.props.book.title : '',
+      description: this.props.book ? this.props.book.description : '',
+      book_url: this.props.book ? this.props.book.bookURL : '',
+      stock_quantity: this.props.book ? this.props.book.stock.quantity : '',
+      stock_record_date: this.props.book ? this.props.book.stock_record_date : '',
+      isbn: this.props.book ? this.props.book.isbn : '',
+      published_date: this.props.book ? this.props.book.published_date : '',
+      author: this.props.book ? this.props.book.author : '',
+      book_category_id: this.props.book ? this.props.book.book_category_id : '',
+      isLoading: false,
+      showCoverBook: false,
+      imageSrc: '',
+      coverPhotoPath: this.props.book ? this.props.book.coverPhotoPath : '',
+      documentPath: this.props.book ? this.props.book.documentPath : '',
+      done: false
+    };
+
+    constructor(props) {
+      super(props);
+      this.onClickSubmit = this.onClickSubmit.bind(this);
+      this.onClickOpenBookCover = this.onClickOpenBookCover.bind(this);
+      this.onChangeUploadInput = this.onChangeUploadInput.bind(this);
+      this.onChangeUploadFile = this.onChangeUploadFile.bind(this);
+      this.onClickOk = this.onClickOk.bind(this);
+    }
+
+    onClickSubmit(event) {
+      event.preventDefault();
+      const { coverPhotoPath } = this.state;
+      this.setState({ isLoading: true });
+      if (coverPhotoPath) {
```

Comment:
## Critical
It's not a good practice to have sensitive data like api and secret keys inside your code. Besides, they are values that can be changed and you don't want to always modify your code for just that.

# A better approach
- Move sensitive information into your `env` file
- Define it inside your webpack so that it can be available globally. use the package [dotenv-webpack](https://github.com/mrsteele/dotenv-webpack)

Reply:
Thank you for the feedback. The file is unused and it has been removed